hail/src/main/scala/is/hail/io/bgen/IndexBgen.scala

```diff
@@ -93,9 +93,8 @@ object IndexBgen {
     val partitioner = new RVDPartitioner(Array("file_idx"), keyType.asInstanceOf[TStruct], rangeBounds)
     val crvd = BgenRDD(hc.sc, partitions, settings, null)
 
-    RVD
-      .coerce(typ, crvd)
-      .repartition(partitioner)
+    RVD.unkeyed(rowType, crvd)
```

Comment:
Does this still sort by locus and alleles? What does `unkeyed` do?

Reply:
`RVD.unkeyed` is the replacement for `UnpartitionedRVD.apply`: it makes an `RVD` with empty key, and doesn't change the `ContextRDD` you pass it. It just allows us to call `RVD` methods on `crvd`.

`rvd.repartition(partitioner, shuffle)` returns an `RVD` whose key is `partitioner`'s key, and whose partitioner is `partitioner`. So by the `RVD` invariant, it must be sorted by the key of `partitioner`, which here is `["file_idx", "locus", "alleles"]`.

If `shuffle = false`, then `rvd` must already have a key starting with `partitioner`'s key, so it is already sorted, and `repartition` doesn't need to shuffle.

`repartition` with `shuffle = true` is the replacement for the old `OrderedRVD.shuffle`.

So, where before the `coerce` did a scan to sample keys, used those to construct a partitioner, shuffled to that partitioner, then repartitioned to the actually wanted partitioner, this will just do one shuffle.